### ISSUE TITLE
showStatusBarBackground property is ignored

### ISSUE THREAD

**Issue by ferusinfo** [Douvi/SlideMenuDovi#14]
Hey - awesome library! :) 

When switching from `1.0.0` to `1.0.1`, the `showStatusBarBackground` property is ignored in function `initView`, leading to always showing a status bar background.

**Comment by Douvi** [comment]
Hey ferusinfo,

Thanks to use my lib. `initView` is totally generic it's here to init all frames and the `SliderMenuAnimation` you want to use. Into `SliderMenuAnimation` there is a method `orderList` (called by `initView` after set all frames) which be use to order all layout. So `showStatusBarBackground`'s value will be check at this point.

If you feel there is a better way to do it, feel free to do a PR with your modification. I will be happy see your solution :).

Thanks

**Comment by ferusinfo** [comment]
Ok, I found what is an issue here.
You are calling the `initView` method in `viewDidLoad` of the `SliderMenuViewController` lifecycle.

When you inherit this VC class and put your `initPanel` method in `viedDidLoad` of the children and put your settings there, you need to put them that **BEFORE** you call `super.viewDidLoad()`.
Otherwise, the settings for things like hiding status bar background will be ignored.

I think this is worth mentioning in the documentation that you need to prepare your settings before calling the `super`. 
Do you agree?

**Comment by ferusinfo** [comment]
Or - I have a better idea - add a required protocol (i.e. `setupSlidingMenuOptions`) where user will make all necessary adjustments and then the protocol method will be called in the `initView` method of the base class?

I can prepare an example in no-time.

**Comment by Douvi** [comment]
```
Ok, I found what is an issue here.
You are calling the initView method in viewDidLoad of the SliderMenuViewController lifecycle.

When you inherit this VC class and put your initPanel method in viedDidLoad of the children and put your settings there, you need to put them that BEFORE you call super.viewDidLoad().
Otherwise, the settings for things like hiding status bar background will be ignored.

I think this is worth mentioning in the documentation that you need to prepare your settings before calling the super. 
Do you agree?
```

I do not recommend you to call `initPanel` before `initView` because any views will be initialised which will be necessary after into `initPanel`.

```
Or - I have a better idea - add a required protocol (i.e. setupSlidingMenuOptions) where user will make all necessary adjustments and then the protocol method will be called in the initView method of the base class?

I can prepare an example in no-time.
```

If you extant `SliderMenuViewController` you will have access to `GlobalVariables` which contain all you need to do any actions. :)

What do you think ?

**Comment by ferusinfo** [comment]
Can you look into my PR? It has the protocol solution, and for people who like to have their VCs logic organized, it could be really helpful.

The problem with GlobalVariables is such that you need a place to setup them - when you have a RootVC class, you want to keep the code regarding it in the View Controller.

> I do not recommend you to call initPanel before initView because any views will be initialised which will be necessary after into initPanel.

You've must read me wrong, I understand that `initView` must happen before  `initPanel` ;)
I just say that using my approach, the call to the protocol method in `initView` is allowing developers to keep the code nice and clean (the just override the method instead of keeping the settings somewhere in other place).

**Comment by Douvi** [comment]
I guess you did not see my post, on you PR. But it's already merge. I will try to realise a new version this week on cocoapods.

You are true I did understand wrong you comment. But after checking your code I totally agree it's will be nice to be able to change on the fly some Menu's options.

For `GlobalVariables`, I will need to setup a documentation, that true. But for the implementation I did not found a better way to do it. Because the menu need ti check the same variables at different points of the process to different class (click, swipe, opening menu, closing menu, etc). So, it was easier to create a protocol with all methods needed with it's own extension to be able to have a single implementation.
